### ISSUE TITLE
Rails/TimeZone doesn't account for Time.new.getgm (Rails)

### ISSUE THREAD

**Issue by parasprite** [rubocop-hq/rubocop#2830]
I could easily be wrong but it looks like getgm is actually just an alias for utc in Rails. http://apidock.com/rails/v4.2.1/ActiveSupport/TimeWithZone/getgm

I'll be switching our code over to use the utc method anyways, but I thought I'd let you know.

```
Time.new.getgm.at_beginning_of_day
     ^^^
```

0.37.0 (using Parser 2.3.0.3, running on ruby 1.9.3 x86_64-linux)

**Comment by attilahorvath** [comment]
I think allowing `Time#utc` in the first place is wrong and goes against the whole purpose of this cop. In my opinion, RuboCop shouldn't allow any pure Ruby `Time` objects to be created instead of the correctly timezone-aware `ActiveSupport::TimeWithZone` objects in Rails code.

To constructs dates and times in Rails, you should always go through the proper interfaces, fetching the timezone you need first, using e.g. `Time#zone`, `Time#zone_default` or `Time#find_zone`.

Using either `Time#utc` or its alias `Time#getgm` will return a pure Ruby `Time` object, same as the dangerous `Time#now` would, which this cop correctly forbids.

Some examples:

```ruby
Time.new.class # Time
Time.now.class # Time
Time.utc.class # Time
Time.getgm.class # Time

Time.zone.now.class # ActiveSupport::TimeWithZone
Time.find_zone('UTC').class # ActiveSupport::TimeWithZone
Time.find_zone('GMT').class # ActiveSupport::TimeWithZone
```

The linked `TimeWithZone#getgm` and the original method, `TimeWithZone#utc` are just there for compatibility with external Ruby code - these objects have to pretend to be plain `Time` objects, so they have to implement the full interface. This doesn't mean that these methods are safe to use in general though.

For this reason I'd propose to forbid all methods that construct pure Ruby `Time` objects. Currently the following two are (in my opinion, wrongly) acceptable by this cop: `Time#utc` and `Time#getlocal`.

**Comment by mikegee** [comment]
The `Rails/TimeZone` cop has a optional configuration of `EnforcedStyle: strict` which provides the behavior you want.

[The documentation](http://rubocop.readthedocs.io/en/latest/cops_rails/#railstimezone) doesn't mention `Time#utc` and `Time#getlocal`, but [the code](https://github.com/bbatsov/rubocop/blob/dba6b75a08c0b09b43e2d7622f169e453b4473a8/lib/rubocop/cop/rails/time_zone.rb#L68-L70) shows that these methods are among the methods accepted by default, but not strictly "safe".
